charmcraft/extensions/twelvefactor.py

```diff
@@ -184,17 +170,41 @@ def get_parts_snippet(self) -> dict[str, Any]:
         """Return the parts to add to parts."""
         return {}
 
+    def get_container_name(self) -> str:
+        """Return name of the container for the app image."""
+        return f"{self.framework}-app"
```

Comment:
Should this maybe be the charm's name? (Likewise `{charm_name}-image` below?)

Reply:
This name is used in other places (like the paas-app-charmer library), and changing it will be a backward incompatible change.

In the new framework (Go), the name is just app.